app/src/main/java/org/ieeeguc/ieeeguc/models/Committee.java

```diff
@@ -93,4 +97,45 @@ public void onResponse(Call call, Response response){
         });
 
     }
+    public void edit(int id , String token , String name , String description , final HTTPResponse HTTP_RESPONSE){
+        OkHttpClient client= new OkHttpClient();
+        JSONObject jsonBody = new JSONObject();
+        try{
+            jsonBody.put("name",name);
+            jsonBody.put("description",description);
+            RequestBody body = RequestBody.create(CONTENT_TYPE, jsonBody.toString());
+        Request request=new Request.Builder()
+                .url("http://ieeeguc.org/api/committee/"+id)
+                .addHeader("Authorization",token)
+                .addHeader("user_agent","Android")
+                .post(body)
+                .build();
+        client.newCall(request).enqueue(new Callback() {
+            public void onFailure(Call call, IOException e) {
+                HTTP_RESPONSE.onFailure(-1,null);
+                call.cancel();
+            }
+            public void onResponse(Call call, okhttp3.Response response) throws IOException {
+                int code=response.code();
+                String c=code+"";
+                String body=response.body().string();
+                try {
+                    JSONObject rr =new JSONObject(body);
+                    if(c.charAt(0)=='2'){
+                        HTTP_RESPONSE.onSuccess(code,rr);
+
+                    }else {
+                        HTTP_RESPONSE.onFailure(code,rr);
+                    }
+                }catch (JSONException e){
+                    HTTP_RESPONSE.onFailure(code,null);
+                }
+
+                response.close();
+            }
+        });
+    }catch (JSONException e){
+            HTTP_RESPONSE.onFailure(-1,null);
```

Comment:
if the error is in the building of the body then the error is not in the connection thus the status code is not -1
However, the status code should be 500 as it is a problem from the application itself